datadog/fwprovider/data_source_datadog_hosts.go

```diff
@@ -0,0 +1,273 @@
+package fwprovider
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV1"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var _ datasource.DataSourceWithConfigure = &hostsDataSource{}
+
+func NewHostsDataSource() datasource.DataSource {
+	return &hostsDataSource{}
+}
+
+type HostListMetadataModel struct {
+	AgentVersion    types.String `tfsdk:"agent_version"`
+	CPU_cores       types.Int64  `tfsdk:"cpu_cores"`
+	Gohai           types.String `tfsdk:"gohai"`
+	Machine         types.String `tfsdk:"machine"`
+	Platform        types.String `tfsdk:"platform"`
+	Processor       types.String `tfsdk:"processor"`
+	PythonV         types.String `tfsdk:"python_version"`
+	Socket_FQDN     types.String `tfsdk:"socket_fqdn"`
+	Socket_Hostname types.String `tfsdk:"socket_hostname"`
+}
+
+type HostListMetricsModel struct {
+	CPU    types.Float64 `tfsdk:"cpu"`
+	IOWait types.Float64 `tfsdk:"iowait"`
+	Load   types.Float64 `tfsdk:"load"`
+}
+
+type HostListModel struct {
+	Aliases          types.List   `tfsdk:"aliases"`
+	Apps             types.List   `tfsdk:"apps"`
+	AWSName          types.String `tfsdk:"aws_name"`
+	HostName         types.String `tfsdk:"host_name"`
+	ID               types.Int64  `tfsdk:"id"`
+	IsMuted          types.Bool   `tfsdk:"is_muted"`
+	LastReportedTime types.Int64  `tfsdk:"last_reported_time"`
+	Meta             types.Object `tfsdk:"meta"`
+	Metrics          types.Object `tfsdk:"metrics"`
+	MuteTimeout      types.Int64  `tfsdk:"mute_timeout"`
+	Name             types.String `tfsdk:"name"`
+	Sources          types.List   `tfsdk:"sources"`
+	Up               types.Bool   `tfsdk:"up"`
+}
+
+type HostsDataSourceModel struct {
+	ID types.String `tfsdk:"id"`
+	// Query Parameters
+	Filter                types.String `tfsdk:"filter"`
+	SortField             types.String `tfsdk:"sort_field"`
+	SortDir               types.String `tfsdk:"sort_dir"`
+	From                  types.Int64  `tfsdk:"from"`
+	IncludeMutedHostsData types.Bool   `tfsdk:"include_muted_hosts_data"`
+	IncludeHostsMetadata  types.Bool   `tfsdk:"include_hosts_metadata"`
```

Comment:
What new attributes does this return? Is it worth exposing this or is it better to just have it always be true?

Reply:
It adds an object with the attributes in `HostListMetadataModel`. When its false, the `Meta` object's attributes are all zero-value. I don't think it would make a huge difference if we made it always true so thats definitely an option.

Reply:
Actually, since the ObjectType requires all fields to be populated I don't think there's much point in this toggle. I'll change it to be on by default, otherwise its just a bunch of empty strings and zero-values.